tests/integration/test_kubernetes_master_integration.py

```diff
@@ -15,6 +15,16 @@ async def test_build_and_deploy(ops_test):
     await ops_test.model.wait_for_idle(wait_for_active=True, timeout=60 * 60)
 
 
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy_pod_security_policy(ops_test):
+    bundle = ops_test.render_bundle(
+        "tests/data/bundle_pod_security_policy.yaml",
+        master_charm=await ops_test.build_charm("."),
+    )
+    await ops_test.model.deploy(bundle)
+    await ops_test.model.wait_for_idle(wait_for_active=True, timeout=60 * 60)
```

Comment:
This test is failing:

```
__________________ test_build_and_deploy_pod_security_policy ___________________
Traceback (most recent call last):
  File "/home/ubuntu/actions-runner/_work/charm-kubernetes-master/charm-kubernetes-master/tests/integration/test_kubernetes_master_integration.py", line 24, in test_build_and_deploy_pod_security_policy
    await ops_test.model.deploy(bundle)
  File "/home/ubuntu/actions-runner/_work/charm-kubernetes-master/charm-kubernetes-master/.tox/integration/lib/python3.8/site-packages/juju/model.py", line 1433, in deploy
    await handler.execute_plan()
  File "/home/ubuntu/actions-runner/_work/charm-kubernetes-master/charm-kubernetes-master/.tox/integration/lib/python3.8/site-packages/juju/bundle.py", line 167, in execute_plan
    self.references[step.id_] = await change.run(self)
  File "/home/ubuntu/actions-runner/_work/charm-kubernetes-master/charm-kubernetes-master/.tox/integration/lib/python3.8/site-packages/juju/bundle.py", line 346, in run
    await context.model._deploy(
  File "/home/ubuntu/actions-runner/_work/charm-kubernetes-master/charm-kubernetes-master/.tox/integration/lib/python3.8/site-packages/juju/model.py", line 1570, in _deploy
    result = await app_facade.Deploy(applications=[app])
  File "/home/ubuntu/actions-runner/_work/charm-kubernetes-master/charm-kubernetes-master/.tox/integration/lib/python3.8/site-packages/juju/client/facade.py", line 480, in wrapper
    reply = await f(*args, **kwargs)
  File "/home/ubuntu/actions-runner/_work/charm-kubernetes-master/charm-kubernetes-master/.tox/integration/lib/python3.8/site-packages/juju/client/_client8.py", line 784, in Deploy
    reply = await self.rpc(msg)
  File "/home/ubuntu/actions-runner/_work/charm-kubernetes-master/charm-kubernetes-master/.tox/integration/lib/python3.8/site-packages/juju/client/facade.py", line 623, in rpc
    result = await self.connection.rpc(msg, encoder=TypeEncoder)
  File "/home/ubuntu/actions-runner/_work/charm-kubernetes-master/charm-kubernetes-master/.tox/integration/lib/python3.8/site-packages/juju/client/connection.py", line 493, in rpc
    raise errors.JujuError(err_results)
juju.errors.JujuError: ['cannot add application "containerd": application already exists']
```

The tests in this module are run in sequence, and they share the same Juju model. So by the time `test_build_and_deploy_security_policy` runs, `test_build_and_deploy` has already deployed the charms into the model.

I think to test this properly, you will need to write a test_pod_security_policy that assumes the charms have already been deployed. I would expect the test to set the kubernetes-master application's pod-security-policy config, and then verify that the pod security policy has been updated in Kubernetes.

Reply:
OK, I am taking this approach and will update this PR.